### PR TITLE
Make `quick-repo-deletion` success message more helpful

### LINKED ISSUE #3984: Improve `quick-repo-deletion`
Screen Shot

On the final information bar:

- [ ] Add link to recovery https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/restoring-a-deleted-repository as suggested by @kidonng 
- [ ] Add link to user’s own links so they can delete more, e.g. https://github.com/fregante?tab=repositories&q=&type=fork&language=

Full example:

```
Repository {url} deleted. You can [restore it](…) or see [your other forks](…).
```

### PR DESCRIPTION
**Pull request by yakov116**
1. LINKED ISSUES:
   Closes #3984 

2. TEST URLS:
None

3. SCREENSHOT:
image

### CODE REVIEW

File: source/features/quick-repo-deletion.tsx [hunk 1]
```diff
@@ -83,12 +83,12 @@ async function start(buttonContainer: HTMLDetailsElement): Promise<void> {
...
 
 	try {
-		const {nameWithOwner} = getRepo()!;
+		const {nameWithOwner, owner} = getRepo()!;
 		await api.v3('/repos/' + nameWithOwner, {
 			method: 'DELETE',
 			json: false
 		});
-		addNotice(`Repository ${nameWithOwner} deleted`, {action: false});
+		addNotice(<span>Repository {nameWithOwner} deleted. You can <a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/restoring-a-deleted-repository">restore it</a> or see <a href={`/${owner}?tab=repositories&q=&type=fork`}>your other forks</a></span>, {action: false});
```
Comment (fregante) [inline source/features/quick-repo-deletion.tsx:91]: Does `owner` also work for organizations? Also maybe we can provide a direct link for recovery if the link is the same for both users and orgs (via `owner` again)

Comment (fregante) [reply source/features/quick-repo-deletion.tsx:91]: I checked now that I'm on the computer. Both URLs vary depending on user vs org:

Deleted repos:

```
https://github.com/settings/deleted_repositories
https://github.com/organizations/GhostText/settings/deleted_repositories
```

Fork list:

```
https://github.com/fregante?tab=repositories&type=fork
https://github.com/GhostText?type=fork
```

The issue is that as far as I know it's not possibly to detect whether a repo belongs to an organization, at least this was the case the last time we needed this piece of information.

For the forks list however, the first actually works for both anyway, `tab` is discarded.

So I think this is ok as is, unless you figure out how to detect an org repo without HTTP calls.

**Comment by fregante** [comment]
> 3\. SCREENSHOT:
>     None

🤷‍♂️

File: source/features/quick-repo-deletion.tsx (hunk 1 above)
Comment (yakov116) [reply source/features/quick-repo-deletion.tsx:91]: $('[data-owner-scoped-search-url]')

Comment (fregante) [reply source/features/quick-repo-deletion.tsx:91]: Great! It's worth adding a `isOrganizationRepo` to detection module. The fork list link can still stay as is though.

Comment (yakov116) [reply source/features/quick-repo-deletion.tsx:91]: I'm vacationing over the weekend, so either I will do it tonight or next week.

**Comment by yakov116** [comment]
Um um um....
"It may take up to an hour for repositories to be displayed here. You can only restore repositories that have no forks or have not been forked"

You cannot restore it (a fork)! 😠

**Comment by fregante** [comment]
We should change our copy to “You might be able to restore it”

File: source/features/quick-repo-deletion.tsx [hunk 2]
```diff
@@ -81,14 +81,14 @@ async function start(buttonContainer: HTMLDetailsElement): Promise<void> {
...
+	const isOrganizationRepo = (): boolean => document.querySelector<HTMLFormElement>('[data-owner-scoped-search-url]')!.dataset.ownerScopedSearchUrl!.startsWith('/org');
 	try {
-		const {nameWithOwner} = getRepo()!;
+		const {nameWithOwner, owner} = getRepo()!;
 		await api.v3('/repos/' + nameWithOwner, {
 			method: 'DELETE',
 			json: false
 		});
-		addNotice(`Repository ${nameWithOwner} deleted`, {action: false});
+		addNotice(<span>Repository {nameWithOwner} deleted. You can <a href={`${isOrganizationRepo() ? '/organizations/' + owner : ''}/settings/deleted_repositories`}>restore it</a> or see <a href={`/${owner}?tab=repositories&q=&type=fork`}>your other forks</a></span>, {action: false});
```
Comment (fregante) [inline source/features/quick-repo-deletion.tsx:91]:
```suggestion
		addNotice(<span>Repository {nameWithOwner} deleted. You might be able to <a href={`${isOrganizationRepo() ? '/organizations/' + owner : ''}/settings/deleted_repositories`}>restore it</a> or see <a href={`/${owner}?tab=repositories&type=fork`}>your other forks</a></span>, {action: false});
```

File: source/features/quick-repo-deletion.tsx [hunk 3]
```diff
@@ -81,14 +81,14 @@ async function start(buttonContainer: HTMLDetailsElement): Promise<void> {
 	}
 
 	select('.btn', buttonContainer)!.textContent = 'Deleting repo…';
-
+	const isOrganizationRepo = (): boolean => document.querySelector<HTMLFormElement>('[data-owner-scoped-search-url]')!.dataset.ownerScopedSearchUrl!.startsWith('/org');
```
Comment (fregante) [inline source/features/quick-repo-deletion.tsx:84]:
```suggestion

```

Just to keep track: the detection package needs to be updated before merging

**Review by fregante — CHANGES_REQUESTED** [review]

File: source/features/quick-repo-deletion.tsx (hunk 3 above)
Comment (fregante) [reply source/features/quick-repo-deletion.tsx:84]: After both changes, it’s good to go

**Pull request merged**